File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        # n = len(nums)
        # if n < 3:
        #     return n
        
        # i = 2
        # for j in range(2,n):
        #     if nums[j] != nums[i-2]:
        #         nums[i] = nums[j]
        #         i += 1
        # return i

        counter = Counter(nums)
        i = 0

        for num, count in counter.items():
            nums[i] = num
            i += 1
            if count > 1:
                nums[i] = num
                i += 1
        return i

File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if(nums.size() <= 2) return nums.size();
        int j=0;
        for( int i=2; i<nums.size(); i++){
            if(nums[i] != nums[j]){
                nums[j+2] = nums[i];
                j++;
            }
        }
        return j+2;
    }
};